### PR TITLE
Request for applying payment method and refreshing totals was added t…

### PR DESCRIPTION
**Pull request by Styopchik**
Request for applying payment method and refreshing totals was added to select-payment-method action. It is necessary for applying shopping cart rules, based on payment method in checkout.

### Description (*)

I've added a request to apply payment method to current quote and recollect totals. It was done for applying cart price rules based on payment method.

### Fixed Issues (if relevant)

1. magento/magento2#24206: Cart price rule based on payment methodis not aplied in checkout

### Manual testing scenarios (*)

1. Create cart price rule based on payment method (f.e. Bank transfer payment)
2. Set discount in rule actions
3. Add product to cart
4. Go to checkout
5. Select Bank transfer payment method
6. Discount appears in totals

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=24207) <br/>All committers have signed the CLA.

File: app/code/Magento/Checkout/view/frontend/web/js/action/select-payment-method.js [hunk 1]
```diff
@@ -7,11 +7,40 @@
...
+        } else {
+            serviceUrl = urlBuilder.createUrl('/guest-carts/:cartId/selected-payment-method', {
+                cartId: quote.getQuoteId()
+            });
+        }
+        fullScreenLoader.startLoader();
+        storage.put(
+            serviceUrl,
+            JSON.stringify(payload)
+        ).success(function () {
```
Comment (fooman) [inline app/code/Magento/Checkout/view/frontend/web/js/action/select-payment-method.js:45]: I believe the call to selected-payment-method can fail. In such cases we would have an endless fullScreenLoader.

Comment (Styopchik) [reply app/code/Magento/Checkout/view/frontend/web/js/action/select-payment-method.js:45]: I understand, Will fix it.

**Review by ghost — COMMENTED** [review]
@Styopchik thanks for making those changes. Having the always with the stop loader looks good to me. I'll leave the rest of the review for someone from the Maintainers. I did not that some of the functional checkout tests are currently failing so those failures would very likely need solving before this can be progressed.

**Comment by Styopchik** [comment]
@fooman, i've checked functional tests results, and i think fails are not related to my changes. I'm not sure how to fix them

**Comment by Styopchik** [comment]
Fixes also #5937

**Review by sidolov — CHANGES_REQUESTED** [review]
Hi @Styopchik , looks like Functional test failures indicate the issue in provided solution. Please, take a look at test results, screenshots, I see that billing address is empty in some cases

**Comment by Styopchik** [comment]
@sidolov, I'm not sure, how changes provided by my commit can affect billing addresses. Could you help me with that?

**Comment by fooman** [comment]
@Styopchik don't have an answer for you at the moment either. However looking at your changes I believe there will be an additional ajax request during the checkout process. The browser might be trying to fill out the billing address while this request is happening, causing it to end up empty. You'll likely need to add an extra step into the failing tests to wait until the payment setting has finished.

**Comment by sidolov** [comment]
@Styopchik probably it would be a good idea go by test steps manually and try to reproduce the issue, then debug the behavior and find the cause.

**Comment by sidolov** [comment]
@Styopchik  I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for the collaboration!

**Pull request closed without merging**